sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -43,11 +43,38 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
 
   protected def checkEvaluation(
       expression: => Expression, expected: Any, inputRow: InternalRow = EmptyRow): Unit = {
-    val catalystValue = CatalystTypeConverters.convertToCatalyst(expected)
+    // No codegen version expects GenericArrayData for array expect Binarytype
+    val catalystValue = expected match {
+      case arr: Array[Byte] if expression.dataType == BinaryType => arr
```

Comment:
I don't see `convertToCatalyst` do special case for `BinaryType` before. Why we need to do that now?